### PR TITLE
chore: add release-please

### PR DESCRIPTION
**Pull request by JGAntunes**
This PR partially addresses #285. The idea is to provide a way to tag, release and bump the versions of both `package.json` and `swagger.yml` files. I tried running some tests on this repo but unfortunately seems like `release-please` will always create PRs against the mainline. As such I've ran some tests yesterday on a small repo - https://github.com/JGAntunes/actions-playground/pull/1 - testing `auto-commit`. The idea here is to rely on it as a workaround to bump the `swagger.yml` file based on `package.json` version (which is initially done by `release-please`).

For this to work I'll need to add a secret to the repo (currently don't have access, maybe you can help me witht it @erezrokah?), similar to what we've done, maybe we can use`netlify-bot`s access token? - https://github.com/netlify/cli/pull/1778#issue-558154582

### CODE REVIEW

**Comment by erezrokah** [comment]
> For this to work I'll need to add a secret to the repo (currently don't have access, maybe you can help me witht it @erezrokah?), similar to what we've done, maybe we can use`netlify-bot`s access token? - [netlify/cli#1778 (comment)](https://github.com/netlify/cli/pull/1778#issue-558154582)

Thanks @JGAntunes I've added the secret. How are we dealing with generating the `go` client?

File: package.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
   "devDependencies": {
     "analytics": "^0.2.0",
     "analytics-plugin-ga": "^0.1.5",
-    "auto-changelog": "^1.16.1",
```
Comment (erezrokah) [inline package.json:20]: I think we're missing the `package-lock.json` files changes

File: .github/workflows/swagger-bump.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  swagger-bump:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm install
```
Comment (erezrokah) [inline .github/workflows/swagger-bump.yml:16]: Should we use `npm ci` here?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Added a few comments, but will delegate to the Backend the rest of the review.

File: .github/workflows/swagger-bump.yml (hunk 2 above)
Comment (JGAntunes) [reply .github/workflows/swagger-bump.yml:16]: I thought `npm ci` only installed `dependencies` and not `devDependencies` (which we need here) but now I'm uncertain - https://docs.npmjs.com/cli/v6/commands/npm-ci - and according to the docs doesn't seem like it. I'll change it, thanks @erezrokah 👍

**Comment by JGAntunes** [comment]
> Thanks @JGAntunes I've added the secret. How are we dealing with generating the go client?

I've added a more extensive comment covering that @erezrokah - https://github.com/netlify/open-api/issues/285#issuecomment-763683630

**Review by mraerino — COMMENTED** [review]
from a process level this looks fine.

this still expects people to manually regenerate the Go client right?

since this has always been the case this is not a blocker: can we think of a process where PRs are denied when the go code is not up-to-date?
that would also block releases that had this problem (but should not if we never merged PRs with failing checks).

**Review by erezrokah — APPROVED** [review]

**Comment by JGAntunes** [comment]
> this still expects people to manually regenerate the Go client right?

@mraerino that's correct yes, I've went through a more lengthy boring explanation here - https://github.com/netlify/open-api/issues/285#issuecomment-763683630 - but essentially we can come out with an action to verify if new files are generated through `make all` and haven't been checked in, and fail the build in that case. I would rather have it as a separate PR though. Maybe later on we can improve the automation for generating the go client, but at least that way we would already have some safeguards in place.

**Review by mraerino — APPROVED** [review]

**Pull request merged**